Mathlib/Data/Polynomial/Derivation.lean

```diff
@@ -0,0 +1,70 @@
+/-
+Copyright (c) 2023 Kevin Buzzard. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kevin Buzzard, Richard Hill
+-/
+import Mathlib.Data.Polynomial.Derivative
+import Mathlib.RingTheory.Derivation.Basic
+import Mathlib.Data.Polynomial.AlgebraMap
+
+/-!
+# Derivations of univariate polynomials
+
+In this file we prove that an `R`-derivation of `Polynomial R` is determined by its value on `X`.
+We also provide a constructor `Polynomial.mkDerivation` that
+builds a derivation from its value on `X`, and a linear equivalence
+`Polynomial.mkDerivationEquiv` between `A` and `Derivation (Polynomial R) A`.
+-/
+
+namespace Polynomial
+
+open scoped BigOperators
+
+noncomputable section
+
+variable {R A : Type _} [CommSemiring R] [AddCommMonoid A] [Module R A] [Module (Polynomial R) A]
+
+section
+
+@[simp]
+theorem derivation_C (D : Derivation R (Polynomial R) A) (a : R) : D (C a) = 0 :=
+  D.map_algebraMap a
+
+-- simp normal form is `derivation_C_mul'`
+theorem derivation_C_mul (D : Derivation R (Polynomial R) A) (a : R) (f : Polynomial R) :
+    D (C a * f) = a • D f := by rw [C_mul', D.map_smul]
+
```

Comment:
This one looks a bit weird to me: it uses simp-normal form on the LHS but not the RHS.

The version with the simp-normal form on both sides is `derivation_C_mul'`, while the one with neither side is `D.map_smul`: so I'd argue we should just drop this lemma.

Reply:
> Should I now rename derivation_C_mul' to derivation_C_mul? Should I make the corresponding change in MvPolynomial?

Yes; and doing it in this PR is fine.

Reply:
OK I've removed the primed lemmas (which were unused in mathlib). Is the name of the lemma even correct though? (sorry, I edited my comment at the same time as you were replying to it :-). That'll teach me.)